File: app/src/main/java/com/recursia/popularmovies/presentation/views/activity/HolderActivity.java
package com.recursia.popularmovies.presentation.views.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.recursia.popularmovies.R;
import com.recursia.popularmovies.Screens;
import com.recursia.popularmovies.TheApplication;

import ru.terrakok.cicerone.Navigator;
import ru.terrakok.cicerone.NavigatorHolder;
import ru.terrakok.cicerone.android.support.SupportAppNavigator;
import ru.terrakok.cicerone.commands.Command;
import ru.terrakok.cicerone.commands.Replace;

public class HolderActivity extends AppCompatActivity {

    private final Navigator navigator = new SupportAppNavigator(this, R.id.main_container) {
        @Override
        public void applyCommands(Command[] commands) {
            super.applyCommands(commands);
            getSupportFragmentManager().executePendingTransactions();
        }
    };
    private NavigatorHolder navigatorHolder;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_holder);

        navigatorHolder = TheApplication
                .getInstance()
                .getAppComponent()
                .getNavigationHolder();

        if (savedInstanceState == null) {
            navigator.applyCommands(new Command[]{new Replace(new Screens.IntroScreen())});
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        navigatorHolder.setNavigator(navigator);
    }

    @Override
    protected void onPause() {
        navigatorHolder.removeNavigator();
        super.onPause();
    }

}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/activity/HolderActivity.kt
package com.recursia.popularmovies.presentation.views.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.recursia.popularmovies.R
import com.recursia.popularmovies.Screens
import com.recursia.popularmovies.TheApplication
import ru.terrakok.cicerone.Navigator
import ru.terrakok.cicerone.NavigatorHolder
import ru.terrakok.cicerone.android.support.SupportAppNavigator
import ru.terrakok.cicerone.commands.Command
import ru.terrakok.cicerone.commands.Replace

class HolderActivity : AppCompatActivity() {
    private val navigator: Navigator = object : SupportAppNavigator(this, R.id.main_container) {
        override fun applyCommands(commands: Array<Command>) {
            super.applyCommands(commands)
            supportFragmentManager.executePendingTransactions()
        }
    }
    private lateinit var navigatorHolder: NavigatorHolder
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_holder)
        navigatorHolder = TheApplication
                .getInstance()
                .appComponent
                .navigationHolder
        if (savedInstanceState == null) {
            navigator.applyCommands(arrayOf<Command>(Replace(Screens.IntroScreen())))
        }
    }

    override fun onResume() {
        super.onResume()
        navigatorHolder.setNavigator(navigator)
    }

    override fun onPause() {
        navigatorHolder.removeNavigator()
        super.onPause()
    }
}
